.github/workflows/driver_ci.yaml

```diff
@@ -0,0 +1,201 @@
+name: CSI IBM Driver CI
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+jobs:
+  prepare_env:
+   name: "prepare env"
+   runs-on: ubuntu-latest
+   steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Setup dependencies
+        run: |
+          scripts/ci/setup_ci_dependencies.sh
+      - uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('dev-requirements.txt') }}
+      - name: Save dependencies files
+        uses: actions/upload-artifact@v2
+        with:
+          name: ci-dependencies
+          path: |
+            /home/runner/.bash_profile
+          retention-days: 1
+      - uses: FranzDiebold/github-env-vars-action@v2
+  CSI-controller-static_code_analysis:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: CSI-controller- static code analysis
+        run: |
+          ./scripts/run_static_code_analysis.sh
+  Unit_testing_coverage:
+    runs-on: ubuntu-latest
+    name: "Unit testing + coverage ${{ matrix.csiblock-component }}"
+    strategy:
+      matrix:
+       include:
+         - csiblock-component: controller
+           command-to-run: "./scripts/run_unitests.sh"
+         - csiblock-component: node
+           command-to-run: "make test-xunit-in-container"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: CSI-${{ matrix.csiblock-component }}- Unit testing + coverage
+        run: |
+          mkdir -p build/reports && chmod 777 build/reports
+          $command_to_run `pwd`/build/reports
+        env:
+          command_to_run: ${{ matrix.command-to-run }}
+
+  CSI-deployment-k8s_yamls_validation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: CSI-deployment- k8s yamls validation
+        run: |
+          ./scripts/run_yamlcheck.sh
+  CSI-Build-and-push-images:
+    runs-on: ubuntu-latest
+    needs:
+      - CSI-controller-static_code_analysis
+      - Unit_testing_coverage
+      - CSI-deployment-k8s_yamls_validation
+    strategy:
+      matrix:
+        image_type: ['node', 'controller']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
```

Comment:
could we use buildx in our regular `build_push_images.sh` script instead?

Reply:
but we still want buildx for internal builds as well - could we utilize these built-in function in our internal builds as well?
or will we have to maintain our own buildx scripts anyway?

Reply:
My planning that we won't use the internal builds because our internal registry doesn't have multi-arch images.
We couldn't utilize this built-in function in our internal builds as well

Reply:
is it because currently our internal registry is at an old version?
because I think it [should work](https://www.jfrog.com/confluence/display/JFROG/Docker+Registry#DockerRegistry-bulk)

Reply:
I remember that we checked and it wasn't support multi-arch

Reply:
I think it should be fixable.
could you please open a ticket for it, and write its number in this conversation?
then we'll keep this as is for now, and in said ticket wilil replace the built-in function usage with the future `build_push_images.sh` buildx implementation